app/models/offer.rb

```diff
@@ -73,6 +77,14 @@ def set_initial_state
       offer.reviewed_at = Time.now
     end
 
+    before_transition :on => [:finish_review, :close] do |offer, transition|
+      offer.review_completed_at = Time.now
+    end
+
+    before_transition :on => :receive do |offer, transition|
```

Comment:
Unused block argument - `transition`. If it's necessary, use `_` or `_transition` as an argument name to indicate that it won't be used.<br>Use the new Ruby 1.9 hash syntax.